### ISSUE TITLE
Cannot create new entries

### ISSUE THREAD

**Issue by ferreria23** [buttercup/buttercup-mobile#316]
I cannot longer create entries in version 2.4.2.

Platform: android 12
Vault type: Dropbox

Steps to reproduce:

1. Try to create a new entry
2. Field the data
3. Try to save
4. Toast show saying failed saving entry, request failed: 0 undefined

**Comment by iaintshootinmis** [comment]
Same error on multiple android devices. Samsung Galaxy 22U and Samsung Galaxy S21 
No issues on Desktop app. 
Dropbox linked vault. 
App version: 2.4.2

**Comment by perry-mitchell** [comment]
Status of `0` is a network failure, usually indicative of no internet access or some other network related fault. Both the mobile app and the desktop use the same URL and library to connect to Dropbox.

I have an Android device that I can test with this week, but so far all the other apps I've tried work fine with Dropbox, including the iOS build.

**Comment by iaintshootinmis** [comment]
This would make sense as I use the PiHole DNS  to limit my exposure to
harmful online material.

However, in this case, it neither works on nor off my network, and the
desktop version is not effected while on network.

Is there a dump feature that would be helpful to provide?

**Comment by perry-mitchell** [comment]
I run PiHoles as well and it works fine- I don't think any of the endpoints are blocked by any common lists.

Unfortunately there's currently no way to dump debug data from the mobile app, but it's a feature we need to add.

Do you have any other programs etc installed on the phone that affect network traffic? Any custom modifications? Does it fail on cellular data as well or just your LAN?

**Comment by jmbothe** [comment]
I am having this exact same issue. I dont run PiHoles, I dont have any other programs that affect network traffic. No Custom mods. Fails on cellular data and LAN.

**Comment by perry-mitchell** [comment]
Unfortunately there's not much I can do unless someone can compile the app locally and try to get some error logs. Without this there's nothing to debug as I can't reproduce. 

Please refrain from me-toos at this point unless you're able to add information that would assist with debugging. If you're able to provide VPN access into a failing network (doubtful anyone would, but it's something to ask) id be very happy to work with you in debugging this remotely.

**Comment by iaintshootinmis** [comment]
@perry-mitchell  If you can hook me up with a debug enabled version or give me some details on how to compile it, I'll be happy too. 

I'm on a win11 box, or I have access to a few debian environments. Please let me know how we can resolve this. I've taught this app to tons of users and have even got my family who isn't tech savvy at all to use this app. They're frustrated and losing hope waiting for a fix. 

(As I missed your question above, no, I don't have any other apps on my phone that effect web traffic. And several Luddite family members who wouldn't now how to modify their dns settings are also affected.) 

Thanks!

**Comment by julianoes** [comment]
Does this output on logcat help at all?

Sync with Dropbox
Android 12
Buttercup 2.4.2

```
08-02 15:38:14.907 19158 19187 E ReactNativeJS: [Layerr: Request failed: 0 undefined]
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: console.error: Layerr: Request failed: 0 undefined, js engine: hermes, stack:
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: _construct@1:195495
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: Wrapper@1:195161
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: _createSuperInternal@1:193103
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: SyntheticError@1:194632
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: reactConsoleErrorHandler@1:194225
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: anonymous@1:3027743
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: tryCallOne@53:15
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: anonymous@139:26
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: anonymous@1:212184
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: _callTimer@1:211080
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: _callReactNativeMicrotasksPass@1:211285
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: callReactNativeMicrotasks@1:213286
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: __callReactNativeMicrotasks@1:117622
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: anonymous@1:116706
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: __guard@1:117463
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: flushedQueue@1:116617
08-02 15:38:14.920 19158 19188 E unknown:ReactNative: callFunctionReturnFlushedQueue@1:116473
```

**Comment by perry-mitchell** [comment]
Ok, seems this is a dupe of #314 so I'll close in favour of that.

I can't reproduce this issue but I am able to reproduce another issue with Dropbox on my android device (can't even add a new vault). I'll try to debug this and get a fix out asap.
